pkg/tsdb/postgres/postgres.go

```diff
@@ -1,87 +1,139 @@
 package postgres
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"reflect"
 	"strconv"
 	"strings"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
 	"github.com/grafana/grafana-plugin-sdk-go/data/sqlutil"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
 	"github.com/grafana/grafana/pkg/setting"
 	"github.com/grafana/grafana/pkg/util/errutil"
 
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/tsdb/sqleng"
 )
 
-func ProvideService(cfg *setting.Cfg) *PostgresService {
-	logger := log.New("tsdb.postgres")
-	return &PostgresService{
+var logger = log.New("tsdb.postgres")
+
+func ProvideService(cfg *setting.Cfg, manager backendplugin.Manager) (*Service, error) {
+	s := &Service{
 		Cfg:        cfg,
-		logger:     logger,
 		tlsManager: newTLSManager(logger, cfg.DataPath),
 	}
+	s.im = datasource.NewInstanceManager(s.newInstanceSettings())
+	factory := coreplugin.New(backend.ServeOpts{
+		QueryDataHandler: s,
+	})
+
+	if err := manager.RegisterAndStart(context.Background(), "postgres", factory); err != nil {
```

Comment:
Think @wbrowne suggested using `Register` for core plugins instead of `RegisterAndStart`.

```suggestion
	if err := manager.Register("postgres", factory); err != nil {
```